tests/upgrade_path/upgrade_helpers.py

```diff
@@ -59,6 +60,40 @@ def check_sonic_version(duthost, target_version):
         "Upgrade sonic failed: target={} current={}".format(target_version, current_version)
 
 
+def restore_image_to_first_boot(duthost, image_version):
+    """
+    Restore the specified image version to a "first boot" state. If the image is installed, and it's not the
+    currently-running image, then the image will be restored to the first boot state. This means that when it boots up,
+    it'll go through platform module installation, container creations, etc. (basically everything that happens when
+    the image first boots up).
+
+    Returns true if the image was successfully restored to first-boot state, false otherwise.
+    """
+    images = duthost.image_facts()['ansible_facts']['ansible_image_facts']
+    if images['current'] == image_version:
+        # We're running the image we want to restore to first boot state, which isn't valid,
+        # so fail here
+        return False
+
+    image_present = False
+    for installed_image in images['available']:
+        if installed_image == image_version:
+            image_present = True
+            break
+    if not image_present:
+        return False
+
+    base_path = os.path.dirname(__file__)
+    restore_to_first_boot_script = os.path.join(base_path, "restoreToFirstBoot.sh")
+    duthost.copy(src=restore_to_first_boot_script, dest="/tmp/restoreToFirstBoot.sh", mode="preserve")
+    duthost.command("sudo unshare -mnpf /tmp/restoreToFirstBoot.sh {}"
```

Comment:
Double checking - will the logic `image_version.replace("SONiC-OS", "image")` work for all versions (public, internal, tagged?)

Reply:
Yes. For example, public is something like `SONiC-OS-master.602773-0324d9acd`.